src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -140,5 +138,34 @@ internal static bool CompletionRequiresQuotes(string completion)
 
         private static bool ContainsCharsToCheck(ReadOnlySpan<char> text)
             => text.ContainsAny(s_defaultCharsToCheck);
+
+        /// <summary>
+        /// Quotes and escapes a given completion text based on the specified options.
```

Comment:
Really we escape only single quote under a condition.
```suggestion
        /// Quotes a given completion text.
```